infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinition.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.restapi.types;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import java.io.IOException;
+
+public class EnumTypeDefinition<T extends Enum<T>> implements DeserializableTypeDefinition<T> {
```

Comment:
Probably worth adding some tests for this.

Reply:
added a serialize test, i think deserialize isn't working currently, will be follow up task.